### PR TITLE
Return partition and nodename as tuple when replying to the read FSM

### PR DESCRIPTION
**Pull request by davecromberge**
When a get request is serviced from cache, the metric vnode responds with information about it's partition, but not the nodename. This information is used by the read FSM to issue commands to repair any data that is in conflict.

This PR fixes/enforces the 2-tuple convention that is required by riak-core, when issuing a command. This is only enforced on a get, since the rest of the commands seem to use a well formed preference list (a list of 2 tuples). The repair scenario seems to be the only case where a destination tuple singleton is used to address a command to a specific VNode directly, by dalmatiner.

### CODE REVIEW

File: apps/metric_vnode/eqc/metric_vnode_eqc.erl [hunk 1]
```diff
@@ -74,19 +74,18 @@ overlap(Tr, Start, Vs) ->
 
 get(S, T, C) ->
     ReqID = T,
+    ReplyNode = nonode@nohost,
     Command = {get, ReqID, ?B, ?M, {T, C}},
     case ?V:handle_command(Command, {raw, ReqID, self()}, S) of
```
Comment (elvisci) [inline apps/metric_vnode/eqc/metric_vnode_eqc.erl]: Don't use macros (like V on line 79) as module names.

**Comment by Licenser** [comment]
Great! 
I know it's somewhat unfair to ask for since the original issue with the check is based on my code but I've started to using elvis (lint) to get the code more readable and it would be nice if the PR would pass those checks.

File: apps/metric_vnode/eqc/metric_vnode_eqc.erl [hunk 2]
```diff
@@ -121,12 +118,12 @@ prop_gb_comp() ->
                    os:cmd("rm -r data"),
                    os:cmd("mkdir data"),
                    {S, T} = eval(D),
-                   List = ?T:to_list(T),
+                   List = gb_trees:to_list(T),
                    List1 = [{get(S, Time, 1), V} || {Time, V} <- List],
                    List2 = [{unlist(mmath_bin:to_list(Vs)), Vt} || {{_ ,Vs}, Vt} <- List1],
```
Comment (elvisci) [inline apps/metric_vnode/eqc/metric_vnode_eqc.erl]: Missing space after "," on line 123

File: apps/metric_vnode/eqc/metric_vnode_eqc.erl [hunk 3]
```diff
@@ -206,21 +203,21 @@ prop_handoff() ->
                    List6 = [true || {_V, _V} <- List2],
 
                    {async,{fold, AsyncHc, _}, _, C2} =
```
Comment (elvisci) [inline apps/metric_vnode/eqc/metric_vnode_eqc.erl]: Missing space after "," on line 205

File: apps/metric_vnode/eqc/metric_vnode_eqc.erl [hunk 4]
```diff
@@ -206,21 +203,21 @@ prop_handoff() ->
...
-                       ?V:handle_handoff_command(FR, self(), C1),
+                       metric_vnode:handle_handoff_command(FR, self(), C1),
                    Lc = AsyncHc(),
                    Lc1 = lists:sort(Lc),
                    {async,{fold, Async, _}, _, _} =
-                       ?V:handle_coverage({metrics, ?B}, all, self(), S1),
+                       metric_vnode:handle_coverage({metrics, ?B}, all, self(), S1),
```
Comment (elvisci) [inline apps/metric_vnode/eqc/metric_vnode_eqc.erl]: Line 210 is too long:                        metric_vnode:handle_coverage({metrics, ?B}, all, self(), S1),.

File: apps/metric_vnode/eqc/metric_vnode_eqc.erl [hunk 5]
```diff
@@ -181,23 +178,23 @@ prop_handoff() ->
                    os:cmd("mkdir data"),
                    {S, T} = eval(D),
 
-                   List = ?T:to_list(T),
+                   List = gb_trees:to_list(T),
 
                    List1 = [{get(S, Time, 1), V} || {Time, V} <- List],
                    List2 = [{unlist(mmath_bin:to_list(Vs)), Vt} || {{_ ,Vs}, Vt} <- List1],
```
Comment (elvisci) [inline apps/metric_vnode/eqc/metric_vnode_eqc.erl]: Line 184 is too long:                    List2 = [{unlist(mmath_bin:to_list(Vs)), Vt} || {{_ ,Vs}, Vt} <- List1],.

File: apps/metric_vnode/eqc/metric_vnode_eqc.erl [hunk 6]
```diff
@@ -181,23 +178,23 @@ prop_handoff() ->
...
                    {async, {fold, AsyncH, _}, _, S1} =
-                       ?V:handle_handoff_command(FR, self(), S),
+                       metric_vnode:handle_handoff_command(FR, self(), S),
                    L = AsyncH(),
                    L1 = lists:sort(L),
-                   {ok, C, _} = ?V:init([1]),
+                   {ok, C, _} = metric_vnode:init([1]),
                    C1 = lists:foldl(fun(Data, SAcc) ->
-                                            {reply, ok, SAcc1} = ?V:handle_handoff_data(Data, SAcc),
+                                            {reply, ok, SAcc1} = metric_vnode:handle_handoff_data(Data, SAcc),
```
Comment (elvisci) [inline apps/metric_vnode/eqc/metric_vnode_eqc.erl]: Line 197 is too long:                                             {reply, ok, SAcc1} = metric_vnode:handle_handoff_data(Data, SAcc),.

File: apps/metric_vnode/eqc/metric_vnode_eqc.erl (hunk 5 above)
Comment (elvisci) [inline apps/metric_vnode/eqc/metric_vnode_eqc.erl]: Missing space after "," on line 184

File: apps/metric_vnode/eqc/metric_vnode_eqc.erl [hunk 7]
```diff
@@ -206,21 +203,21 @@ prop_handoff() ->
                    List6 = [true || {_V, _V} <- List2],
 
                    {async,{fold, AsyncHc, _}, _, C2} =
-                       ?V:handle_handoff_command(FR, self(), C1),
+                       metric_vnode:handle_handoff_command(FR, self(), C1),
                    Lc = AsyncHc(),
                    Lc1 = lists:sort(Lc),
                    {async,{fold, Async, _}, _, _} =
```
Comment (elvisci) [inline apps/metric_vnode/eqc/metric_vnode_eqc.erl]: Missing space after "," on line 209

File: apps/metric_vnode/eqc/metric_vnode_eqc.erl (hunk 2 above)
Comment (elvisci) [inline apps/metric_vnode/eqc/metric_vnode_eqc.erl]: Line 123 is too long:                    List2 = [{unlist(mmath_bin:to_list(Vs)), Vt} || {{_ ,Vs}, Vt} <- List1],.

File: apps/metric_vnode/eqc/metric_vnode_eqc.erl [hunk 8]
```diff
@@ -206,21 +203,21 @@ prop_handoff() ->
...
-                       ?V:handle_coverage({metrics, ?B}, all, self(), S1),
+                       metric_vnode:handle_coverage({metrics, ?B}, all, self(), S1),
                    Ms = Async(),
                    {async,{fold, AsyncC, _}, _, _} =
-                       ?V:handle_coverage({metrics, ?B}, all, self(), C2),
+                       metric_vnode:handle_coverage({metrics, ?B}, all, self(), C2),
```
Comment (elvisci) [inline apps/metric_vnode/eqc/metric_vnode_eqc.erl]: Line 213 is too long:                        metric_vnode:handle_coverage({metrics, ?B}, all, self(), C2),.

File: apps/metric_vnode/eqc/metric_vnode_eqc.erl [hunk 9]
```diff
@@ -206,21 +203,21 @@ prop_handoff() ->
...
                    {async,{fold, AsyncHc, _}, _, C2} =
-                       ?V:handle_handoff_command(FR, self(), C1),
+                       metric_vnode:handle_handoff_command(FR, self(), C1),
                    Lc = AsyncHc(),
                    Lc1 = lists:sort(Lc),
                    {async,{fold, Async, _}, _, _} =
-                       ?V:handle_coverage({metrics, ?B}, all, self(), S1),
+                       metric_vnode:handle_coverage({metrics, ?B}, all, self(), S1),
                    Ms = Async(),
                    {async,{fold, AsyncC, _}, _, _} =
```
Comment (elvisci) [inline apps/metric_vnode/eqc/metric_vnode_eqc.erl]: Missing space after "," on line 212

File: apps/metric_vnode/eqc/metric_vnode_eqc.erl [hunk 10]
```diff
@@ -145,15 +142,15 @@ prop_is_empty() ->
...
+                   TreeEmpty = gb_trees:is_empty(T),
                    if
                        Empty == TreeEmpty ->
                            ok;
                        true ->
                            io:format(user, "~p == ~p~n", [S, T])
                    end,
-                   ?V:terminate(normal, S),
+                   metric_vnode:terminate(normal, S),
                    ?WHENFAIL(io:format(user, "L: ~p /= ~p~n", [Empty, TreeEmpty]),
```
Comment (elvisci) [inline apps/metric_vnode/eqc/metric_vnode_eqc.erl]: Line 154 is too long:                    ?WHENFAIL(io:format(user, "L: ~p /= ~p~n", [Empty, TreeEmpty]),.

File: apps/metric_vnode/eqc/metric_vnode_eqc.erl [hunk 11]
```diff
@@ -206,21 +203,23 @@ prop_handoff() ->
...
-                       ?V:handle_handoff_command(FR, self(), C1),
+                       metric_vnode:handle_handoff_command(FR, self(), C1),
                    Lc = AsyncHc(),
                    Lc1 = lists:sort(Lc),
                    {async,{fold, Async, _}, _, _} =
-                       ?V:handle_coverage({metrics, ?B}, all, self(), S1),
+                       metric_vnode:handle_coverage({metrics, ?B},
+                                                    all, self(), S1),
                    Ms = Async(),
                    {async,{fold, AsyncC, _}, _, _} =
```
Comment (elvisci) [inline apps/metric_vnode/eqc/metric_vnode_eqc.erl]: Missing space after "," on line 213

**Comment by davecromberge** [comment]
@Licenser the checks pass locally, but the review failed with a bad match. Can you take a look?

**Comment by Licenser** [comment]
thanks mate!

**Pull request merged**